### ISSUE TITLE
Throttling: Rate exceeded error when reconcile ALB

### ISSUE THREAD

**Issue by thanhma** [kubernetes-sigs/aws-load-balancer-controller#1871]
I'm getting the Rate exceeded error in v2.1.3
```
{"level":"error","ts":1615309687.1415963,"logger":"controller","msg":"Reconciler error","reconcilerGroup":"elbv2.k8s.aws","reconcilerKind":"TargetGroupBinding","controller":"targetGroupBinding","name":"k8s-default-alias4bh-d5c9e3xxxx","namespace":"default","error":"Throttling: Rate exceeded\n\tstatus code: 400, request id: 4eef24db-b4b3-46bf-88da-5ed3eb4xxxxx"}
```
The `targetgroupbindingMaxConcurrentReconciles` is left to None (as by default).

After query the `ThrottlingException` as guided in https://aws.amazon.com/premiumsupport/knowledge-center/cloudtrail-rate-exceeded/, I found that exception thrown on `DescribeTargetHealth`, `DescribeTargetGroups`, `DescribeTags`, `RegisterTargets`... events

The number of ingress is big (about 10 ingresses per cluster, 200 ingresses in a single VPC). Target type is `instance`, each cluster has about 60 instances and the number of target per ingress may reach to 5,000 since rules are numerous as well.

**Comment by M00nF1sh** [comment]
@thanhma 
By default, the controller didn't set any throttling for the AWS API calls, and we provide a flag to set it by customers.
let me find the recommended settings and reply back here

**Comment by thanhma** [comment]
@M00nF1sh 
In CloudTrail log, I can see many `DescribeTags`, `DescribeTargetHealth`, `DescribeTargetGroups` calls getting `ThrottlingException`. Do these failures affect reconcile with `instance` target type? May it cause target register failed, or performance degrade?

Do we have a rate limit configuration?

**Comment by thanhma** [comment]
@M00nF1sh 
My cluster stopped reconciling ALB because of `RegisterTarget` facing `Throttling Exception: Rate exceeded` error. 
Is there a way to optimize API calls to prevent throttling?

**Comment by M00nF1sh** [comment]
@thanhma 

Yes, you can set via controller flag: ```--aws-api-throttle=Elastic Load Balancing v2:^RegisterTarget=20:3,Elastic Load Balancing v2:^Describe=40:10```
Above is the default throttle for normal AWS accounts. If you have multiple controller/cluster, they are sharing the same throttle per account/region, and you'll need to lower the throttle settings above.

**Comment by ajaykumarmandapati** [comment]
Hi @M00nF1sh, we do have a similar issue and face throttling exceptions as well, could you please direct us on how exactly to configure `--aws-api-throttle` parameter? 
We also found this documentation [here](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.2/deploy/configurations/#default-throttle-config) but could not relate to it since we already disabled `waf` , `waf2` and `shield`. 
Thanks in advance!

**Comment by thanhma** [comment]
@ajaykumarmandapati I found setting the `api-throttle` quite difficult if we don't know when the API throttle reached (how many calls per second is acceptable).

However, in v2.2.0, you can use `alb.ingress.kubernetes.io/target-node-labels` to limit the targets in a single TargetGroup, thus lower the chance API hits the throttle.

**Comment by lmbaschiera** [comment]
I'm having this same issue with v2.4.4 after upgrading from v1.2.0-alpha.1
Hundreds of `DescribeTargetHealth`, `DescribeTargetGroups`, `DescribeTags` and `RegisterTargets` events are making us hit the API rate limits impacting all clusters - even the ones not upgraded to the new controller version.

How can I fix this issue? I've asked for an API rate limit increase but that's just a band-aid for this issue.

**Comment by oliviassss** [comment]
@thanhma The RGT API support has been shipped with the v2.5.2 release, you could enable this feature by the feature gate flag `EnableRGTAPI` to avoid the throttling issue. Please refer to the [release note](https://github.com/kubernetes-sigs/aws-load-balancer-controller/releases/tag/v2.5.2) for more details. I'm closing this issue as of now.  Please feel free to reach out if you're still having this issue.

**Comment by oliviassss** [comment]
@thanhma The RGT API support has been shipped with the v2.5.2 release, you could enable this feature by the feature gate flag EnableRGTAPI to avoid the throttling issue. Please refer to the [release note](https://github.com/kubernetes-sigs/aws-load-balancer-controller/releases/tag/v2.5.2) for more details.
